src/msvc_helper_main-win32.cc

```diff
@@ -113,12 +113,12 @@ int MSVCHelperMain(int argc, char** argv) {
     PushPathIntoEnvironment(env);
   }
 
-  char* command = GetCommandLineA();
-  command = strstr(command, " -- ");
-  if (!command) {
+  std::string utf8_command_line = WideToUtf8(GetCommandLine());
+  std::size_t command_pos = utf8_command_line.find(" -- ");
+  if(command_pos == std::string::npos){
     Fatal("expected command line to end with \" -- command args\"");
   }
-  command += 4;
+  std::string command = utf8_command_line.substr(command_pos + 4);
```

Comment:
When you adjust this code to operate on wchar_t*, make sure you only convert the absolute minimum amount of string data from wchar_t to utf8.

Reply:
@jlonnberg